diff -ruN PsiCLASS-1.0.3_orig/Makefile PsiCLASS-1.0.3/Makefile
--- PsiCLASS-1.0.3_orig/Makefile	2022-09-09 14:47:52.343258000 +0200
+++ PsiCLASS-1.0.3/Makefile	2022-09-13 16:13:47.436239000 +0200
@@ -1,7 +1,7 @@
 CXX = g++
 CXXFLAGS= -Wall -O3 #-g #-std=c++11 #-Wall #-g
 #CXXFLAGS= -Wall -g #-std=c++11 #-Wall #-g
-LINKPATH= -I./samtools-0.1.19 -L./samtools-0.1.19
+LINKPATH= -I$(EBROOTBAMTOOLS)/include/bamtools/ -L$(EBROOTBAMTOOLS)/lib/ -I./samtools-0.1.19 -L./samtools-0.1.19
 LINKFLAGS = -lbam -lz -lm -lpthread 
 DEBUG=
 OBJECTS = stats.o subexon-graph.o 
