--- h5reader.c.orig	2025-12-11 17:23:51.186652000 +0100
+++ h5reader.c	2025-12-11 17:29:41.996299000 +0100
@@ -3313,7 +3313,7 @@
  *************************************************************************
  */
 
-herr_t _NclHDF5check_attr(hid_t obj_id, char *attr_name, const H5A_info_t *ainfo,
+herr_t _NclHDF5check_attr(hid_t obj_id, const char *attr_name, const H5A_info_t *ainfo,
                           void *attr_data)
 {
     hid_t       attr_id, space, type, p_type;
@@ -3985,7 +3985,7 @@
  *************************************************************************
  */
 
-herr_t _NclHDF5search_obj(char *name, H5O_info_t *oinfo,
+herr_t _NclHDF5search_obj(const char *name, const H5O_info_t *oinfo,
                           void *_NclHDF5group, char *already_seen)
 {
     NclHDF5attr_list_t  *curAttrList = NULL;
@@ -4349,7 +4349,7 @@
  *************************************************************************
  */
 
-herr_t _NclHDF5search_link(char *name, H5O_info_t *oinfo, void *_NclHDF5Rec)
+herr_t _NclHDF5search_link(const char *name, const H5L_info_t *linfo, void *_NclHDF5Rec)
 {
 #if DEBUG_NCL_HDF5
     fprintf(stderr, "\nEntering _NclHDF5search_link, name = <%s>, at file: %s, line: %d\n", name, __FILE__, __LINE__);
