--- ni/src/ncl/NclHDF5.c.orig	2025-11-27 14:44:44.163841000 +0100
+++ ni/src/ncl/NclHDF5.c	2025-11-27 14:45:12.666608000 +0100
@@ -5742,6 +5742,7 @@
 /* NclGetGrpAttNamesFunc    get_grp_att_names; */  HDF5GetGrpAttNames,
 /* NclGetGrpAttInfoFunc     get_grp_att_info; */   HDF5GetGrpAttInfo,
 /* NclAddGrpFunc           add_grp; */                  NULL,
+/* NclAddVlenFunc           add_vlen; */           NULL,
 /* NclSetOptionFunc         set_option;  */        HDF5SetOption
 };
 
