--- test_dtype.py.orig	2026-05-28 12:49:29.328293000 +0200
+++ test_dtype.py	2026-05-28 12:50:37.937955000 +0200
@@ -981,10 +981,15 @@
     @pytest.mark.skipif(IS_PYSTON, reason="Pyston disables recursion checking")
     @pytest.mark.skipif(IS_WASM, reason="Pyodide/WASM has limited stack size")
     def test_list_recursion(self):
-        l = []
-        l.append(('f', l))
-        with pytest.raises(RecursionError):
-            np.dtype(l)
+        old_limit = sys.getrecursionlimit()
+        try:
+            sys.setrecursionlimit(100)
+            l = []
+            l.append(('f', l))
+            with pytest.raises(RecursionError):
+                np.dtype(l)
+        finally:
+            sys.setrecursionlimit(old_limit)
 
     @pytest.mark.skipif(IS_PYSTON, reason="Pyston disables recursion checking")
     @pytest.mark.skipif(IS_WASM, reason="Pyodide/WASM has limited stack size")
