Update NclNetCdf.c to modern headers and stricter diagnostics
Author: Pavel Tomanek (Inuits)
--- ni/src/ncl/NclNetCdf.c.orig	2025-11-26 14:18:53.679177000 +0100
+++ ni/src/ncl/NclNetCdf.c	2025-11-26 14:20:38.004171000 +0100
@@ -251,7 +251,7 @@
 			*(nc_type*)out_type = NC_LONG;
 		} else {
 			NhlPError(NhlWARNING,NhlEUNKNOWN,"Classic NetCDF does not support 8-byte longs; convert to a supported data type or use the NetCDF4 format");
-			*(nc_type*)out_type = NULL;
+			*(nc_type*)out_type = NC_NAT;
 		}
 		break;
 	case NCL_float:
