--- makefile.orig	2025-01-30 11:03:58.776449332 +0100
+++ makefile	2025-01-30 11:06:16.334240540 +0100
@@ -3,13 +3,13 @@
 CPPFLAGS=-Wall -Wextra -std=c++17 -O3 -g -Iconcurrentqueue -IBBHash -Izstr/src -Iparallel-hashmap/parallel_hashmap/ `pkg-config --cflags protobuf` `pkg-config --cflags libsparsehash` -Wno-unused-parameter -IMEMfinder/src -I`jemalloc-config --includedir` -Icxxopts/include
 # silly workaround: bamtools does not have pkg-config cflags for finding the include directory
 # instead assume it's a folder at the same location as zlib
-CPPFLAGS+=`pkg-config --cflags zlib`/bamtools
+CPPFLAGS += -I$(EBROOTBAMTOOLS)/include/bamtools
 
 ODIR=obj
 BINDIR=bin
 SRCDIR=src
 
-LIBS=-lm -lz `pkg-config --libs protobuf` -lsdsl -lbamtools
+LIBS=-lm -lz -lsdsl -lbamtools -L$(EBROOTPROTOBUF)/lib -L$(EBROOTABSEIL)/lib
 JEMALLOCFLAGS= -L`jemalloc-config --libdir` -Wl,-rpath,`jemalloc-config --libdir` -ljemalloc `jemalloc-config --libs`
 
 _DEPS = vg.pb.h fastqloader.h GraphAlignerWrapper.h vg.pb.h BigraphToDigraph.h stream.hpp Aligner.h ThreadReadAssertion.h AlignmentGraph.h CommonUtils.h GfaGraph.h ReadCorrection.h MinimizerSeeder.h AlignmentSelection.h EValue.h MEMSeeder.h DNAString.h DiploidHeuristic.h
