--- gromacs-constantph/src/gromacs/listed_forces/gpubondedkernels.cu.orig	2026-01-19 12:36:26.407571000 +0100
+++ gromacs-constantph/src/gromacs/listed_forces/gpubondedkernels.cu	2026-01-19 12:38:40.239178000 +0100
@@ -129,8 +129,8 @@
             fbond *= rsqrtf(dr2);
 
             float3 fij = fbond * dx;
-            atomicAdd(&gm_f[ai], fij);
-            atomicAdd(&gm_f[aj], -fij);
+            gmxAtomicAdd(&gm_f[ai], fij);
+            gmxAtomicAdd(&gm_f[aj], -fij);
             if (calcVir && ki != CENTRAL)
             {
                 atomicAdd(&sm_fShiftLoc[ki], fij);
@@ -217,9 +217,9 @@
             float3 f_k = ckk * r_kj - cik * r_ij;
             float3 f_j = -f_i - f_k;
 
-            atomicAdd(&gm_f[ai], f_i);
-            atomicAdd(&gm_f[aj], f_j);
-            atomicAdd(&gm_f[ak], f_k);
+            gmxAtomicAdd(&gm_f[ai], f_i);
+            gmxAtomicAdd(&gm_f[aj], f_j);
+            gmxAtomicAdd(&gm_f[ak], f_k);
 
             if (calcVir)
             {
@@ -299,9 +299,9 @@
             float3 f_k = ckk * r_kj - cik * r_ij;
             float3 f_j = -f_i - f_k;
 
-            atomicAdd(&gm_f[ai], f_i);
-            atomicAdd(&gm_f[aj], f_j);
-            atomicAdd(&gm_f[ak], f_k);
+            gmxAtomicAdd(&gm_f[ai], f_i);
+            gmxAtomicAdd(&gm_f[aj], f_j);
+            gmxAtomicAdd(&gm_f[ak], f_k);
 
             if (calcVir)
             {
@@ -322,8 +322,8 @@
             fbond *= rsqrtf(dr2);
 
             float3 fik = fbond * r_ik;
-            atomicAdd(&gm_f[ai], fik);
-            atomicAdd(&gm_f[ak], -fik);
+            gmxAtomicAdd(&gm_f[ai], fik);
+            gmxAtomicAdd(&gm_f[ak], -fik);
 
             if (calcVir && ki != CENTRAL)
             {
@@ -417,10 +417,10 @@
         float3 f_j  = f_i - svec;
         float3 f_k  = f_l + svec;
 
-        atomicAdd(&gm_f[i], f_i);
-        atomicAdd(&gm_f[j], -f_j);
-        atomicAdd(&gm_f[k], -f_k);
-        atomicAdd(&gm_f[l], f_l);
+        gmxAtomicAdd(&gm_f[i], f_i);
+        gmxAtomicAdd(&gm_f[j], -f_j);
+        gmxAtomicAdd(&gm_f[k], -f_k);
+        gmxAtomicAdd(&gm_f[l], f_l);
 
         if (calcVir)
         {
@@ -699,8 +699,8 @@
         float3 f     = finvr * dr;
 
         /* Add the forces */
-        atomicAdd(&gm_f[ai], f);
-        atomicAdd(&gm_f[aj], -f);
+        gmxAtomicAdd(&gm_f[ai], f);
+        gmxAtomicAdd(&gm_f[aj], -f);
 
         /* Calculate the 1-4 forces (to enable constant-pH bonded on GPU) */
         float force_14_ai = scale_factor * gm_xq[aj].w * rinv;
