--- setup.py.orig	2024-04-02 18:56:13.015206000 +0200
+++ setup.py	2024-04-02 19:03:24.834334000 +0200
@@ -262,23 +262,27 @@
 cython_msh = Extension(
     'simnibs.mesh_tools.cython_msh',
     ["simnibs/mesh_tools/cython_msh.pyx"],
-    include_dirs=[np.get_include()]
+    include_dirs=[np.get_include()],
+    define_macros=[('NPY_NO_DEPRECATED_API', 'NPY_1_7_API_VERSION')],
 )
 marching_cubes_lewiner_cy = Extension(
     'simnibs.segmentation._marching_cubes_lewiner_cy',
     ["simnibs/segmentation/_marching_cubes_lewiner_cy.pyx"],
-    include_dirs=[np.get_include()]
+    include_dirs=[np.get_include()],
+    define_macros=[('NPY_NO_DEPRECATED_API', 'NPY_1_7_API_VERSION')],
 )
 cat_c_utils = Extension(
     'simnibs.segmentation._cat_c_utils',
     ["simnibs/segmentation/_cat_c_utils.pyx", "simnibs/segmentation/cat_c_utils/genus0.c"],
     include_dirs=[np.get_include(), 'simnibs/segmentation/cat_c_utils'],
-    extra_compile_args=cat_compile_args
+    extra_compile_args=cat_compile_args,
+    define_macros=[('NPY_NO_DEPRECATED_API', 'NPY_1_7_API_VERSION')],
 )
 thickness = Extension(
     'simnibs.segmentation._thickness',
     ["simnibs/segmentation/_thickness.pyx"],
-    include_dirs=[np.get_include()]
+    include_dirs=[np.get_include()],
+    define_macros=[('NPY_NO_DEPRECATED_API', 'NPY_1_7_API_VERSION')],
 )
 petsc_solver = Extension(
     'simnibs.simulation.petsc_solver',
@@ -510,13 +514,5 @@
           'cython'
       ],
       tests_require=['pytest', 'mock'],
-      zip_safe=False)
-
-
-script_dir = shutil.which('simnibs')
-if script_dir is None:
-    raise IOError('could not locate folder with console-scripts')
-else:
-    script_dir = os.path.dirname(script_dir)
-    add_symlinks_or_cmd(external_progs,script_dir)
-    
\ No newline at end of file
+      zip_safe=False
+)
