--- ni/src/ncl/BuiltInFuncs.c.orig	2025-12-05 14:19:14.752125121 +0100
+++ ni/src/ncl/BuiltInFuncs.c	2025-12-05 14:37:24.557839431 +0100
@@ -4069,7 +4069,7 @@
 	}
 	if (ind1 <= 0) {
 		NhlPError(NhlFATAL,NhlEUNKNOWN,"fbinread: 0 or less than zero fortran control word, FILE NOT SEQUENTIAL ACCESS!");
-		close(fd);
+		fclose(fd);
 		return(NhlFATAL);
 	}
 	if(tmp_size == -1) {
@@ -4083,7 +4083,7 @@
 		if (totalsize > ind1) {
 			NhlPError(NhlFATAL,NhlEUNKNOWN,
 				  "fbinread: requested variable size exceeds record size");
-			close(fd);
+			fclose(fd);
 			return(NhlFATAL);
 		}
 	}
