diff -u pod5-file-format-0.1.16/CMakeLists.txt.orig pod5-file-format-0.1.16/CMakeLists.txt
--- pod5-file-format-0.1.16/CMakeLists.txt.orig	2023-01-20 12:22:05.000000000 +0100
+++ pod5-file-format-0.1.16/CMakeLists.txt	2023-01-23 10:34:44.860294168 +0100
@@ -92,7 +91,6 @@

 if (BUILD_PYTHON_WHEEL)
     find_package (Python ${PYTHON_VERSION} EXACT COMPONENTS Interpreter Development)
-    add_subdirectory(third_party/pybind11)
 endif()

 add_subdirectory(c++)
diff -u pod5-file-format-0.1.16/pyproject.toml.orig pod5-file-format-0.1.16/pyproject.toml
--- pod5-file-format-0.1.16/pyproject.toml.orig 2023-04-06 12:50:32.000000000 +0200
+++ pod5-file-format-0.1.16/pyproject.toml      2023-04-18 09:43:20.483040771 +0200
@@ -2,7 +2,7 @@
 # To create this file, simply pip install the root pod-file-format directory

 [build-system]
-requires = ["setuptools >= 61.0", "wheel", "setuptools_scm[toml]>=6.2"]
+requires = ["setuptools >= 61.0", "wheel"]
 build-backend = "setuptools.build_meta"

 [project]
@@ -14,7 +14,3 @@
 [tool.setuptools.packages.find]
 # Ignore all files
 exclude = ["*"]
-
-[tool.setuptools_scm]
-root = "."
-write_to = "_version.py"
diff -u pod5-file-format-0.1.16/python/pod5/pyproject.toml.orig pod5-file-format-0.1.16/python/pod5/pyproject.toml
--- pod5-file-format-0.1.16/python/pod5/pyproject.toml.orig     2023-04-06 12:50:32.000000000 +0200
+++ pod5-file-format-0.1.16/python/pod5/pyproject.toml  2023-04-18 13:59:52.055854106 +0200
@@ -2,7 +2,6 @@
 requires = [
   "setuptools >= 61.0",
   "wheel",
-  "setuptools_scm[toml]>=6.2",
 ]
 build-backend = "setuptools.build_meta"

@@ -63,8 +62,5 @@
 where = ["src"]
 exclude = ["test*"]

-[tool.setuptools_scm]
-root = "../.."
-
 [tool.black]
 target-version = ["py37"]
