diff --git a/MODULE.bazel b/MODULE.bazel
--- a/MODULE.bazel
+++ b/MODULE.bazel
@@ -30,7 +30,6 @@
 bazel_dep(name = "xla")
 archive_override(
     module_name = "xla",
-    integrity = "sha256-SrTyccoJ+p5DG5r13tgj/PA5MtlK2Y9pV/E1I/U7oHA=",
     strip_prefix = "xla-9b635916ecc6df6efee62d8e4b0c7ef87ef84d69",
     urls = ["https://github.com/openxla/xla/archive/9b635916ecc6df6efee62d8e4b0c7ef87ef84d69.tar.gz"],
 )