--- mlir/include/mlir/IR/BuiltinDialectBytecode.td.orig 2026-06-16 14:29:40.248784001 +0200
+++ mlir/include/mlir/IR/BuiltinDialectBytecode.td      2026-06-16 14:30:43.394847510 +0200
@@ -124,7 +124,7 @@
   // FusedLoc::get(context, locs) may return UnknownLoc when locs is empty.
   // Pass Attribute() explicitly to bypass the folding logic and always
   // construct a FusedLoc via Base::getChecked.
-  let cBuilder = "cast<FusedLoc>(getChecked<FusedLoc>([&]() { return reader.emitError(); }, context, $_args, Attribute()))";
+  let cBuilder = "cast<FusedLoc>(get<FusedLoc>(context, $_args, Attribute()))";
 }
 
 def FusedLocWithMetadata : DialectAttribute<(attr
@@ -132,7 +132,7 @@
   Attribute:$metadata
 )> {
   let printerPredicate = "$_val.getMetadata()";
-  let cBuilder = "cast<FusedLoc>(getChecked<FusedLoc>([&]() { return reader.emitError(); }, context, $_args))";
+  let cBuilder = "cast<FusedLoc>(get<FusedLoc>(context, $_args))";
 }
 }
 
@@ -148,7 +148,7 @@
   ResourceHandle<"DenseResourceElementsHandle">:$rawHandle
 )> {
   // Note: order of serialization does not match order of builder.
-  let cBuilder = "getChecked<$_resultType>([&]() { return reader.emitError(); }, context, type, *rawHandle)";
+  let cBuilder = "get<$_resultType>(context, type, *rawHandle)";
 }
 
 let cType = "RankedTensorType" in {