--- setup.py.orig	2025-06-12 16:04:07.452753479 +0200
+++ setup.py	2025-06-12 16:04:52.757963528 +0200
@@ -64,5 +64,9 @@
     packages=['MorphoSONIC'],
     scripts=getFiles('scripts') + getFiles('tests') + getFiles('examples'),
     install_requires=get_requirements(),
-    zip_safe=False
+    zip_safe=False,
+    include_package_data=True,
+    package_data={
+        'MorphoSONIC': ['nmodl/*', 'nmodl/x86_64/.libs/*'],
+    },
 )
\ No newline at end of file
