diff -ruN scib-pipeline-main_orig/envs/create_conda_environments.sh scib-pipeline-main/envs/create_conda_environments.sh
--- scib-pipeline-main_orig/envs/create_conda_environments.sh	2022-10-14 19:37:05.000000000 +0200
+++ scib-pipeline-main/envs/create_conda_environments.sh	2023-02-07 16:14:46.468039486 +0100
@@ -66,7 +66,7 @@
 
   if [ "$QUIET" != "-q" ]; then
     # list all pip dependencies
-    conda run -n $env pip list
+    pip list
   fi
 
   # Set environment variables
@@ -88,11 +88,9 @@
 echo "Install R packages through R"
 
 echo "scib pipeline environment..."
-conda run -n "scib-pipeline-R${R_VERSION}" --no-capture-output \
- Rscript -e "remotes::install_github('theislab/kBET', quiet=TRUE)"
+Rscript -e "remotes::install_github('theislab/kBET', quiet=TRUE)"
 
 echo "scib R environment..."
-conda run -n "scib-R${R_VERSION}" --no-capture-output \
- Rscript "$SCRIPT_DIR/install_R_methods.R" -d "$SCRIPT_DIR/dependencies-R${R_VERSION}.tsv" $QUIET
+Rscript "$SCRIPT_DIR/install_R_methods.R" -d "$SCRIPT_DIR/dependencies-R${R_VERSION}.tsv" $QUIET
 
 echo "Done."
diff -ruN scib-pipeline-main_orig/scripts/integration/Snakefile scib-pipeline-main/scripts/integration/Snakefile
--- scib-pipeline-main_orig/scripts/integration/Snakefile	2022-10-14 19:37:05.000000000 +0200
+++ scib-pipeline-main/scripts/integration/Snakefile	2023-02-07 16:15:08.120040593 +0100
@@ -21,7 +21,7 @@
         batch_key = lambda wildcards: cfg.get_from_scenario(wildcards.scenario, key="batch_key"),
         cell_type = cfg.get_celltype_option_for_integration,
         hvgs      = cfg.get_hvg,
-        cmd       = f"conda run --no-capture-output -n {cfg.py_env} python"
+        cmd       = f"python"
     benchmark:
         str(cfg.ROOT / "{scenario}/integration/{scaling}/{hvg}/{method}.h5ad.benchmark")
     shell:
@@ -51,7 +51,7 @@
     params:
         batch_key = lambda w: cfg.get_from_scenario(w.scenario, key="batch_key"),
         hvgs      = lambda w: cfg.get_hvg(w, rules.integration_prepare.output, prep="RDS"),
-        cmd       = f"conda run --no-capture-output -n {cfg.r_env} Rscript",
+        cmd       = f"Rscript",
     benchmark:
         str(cfg.ROOT / "{scenario}/integration/{scaling}/{hvg}/R/{method}.RDS.benchmark")
     shell:
@@ -76,7 +76,7 @@
         Convert integrated data from {wildcards.method} into h5ad
         """
     params:
-        cmd = f"conda run --no-capture-output -n {cfg.py_env} python"
+        cmd = f"python"
     shell:
         """
         if [ {wildcards.method} == "conos" ]
diff -ruN scib-pipeline-main_orig/scripts/metrics/Snakefile scib-pipeline-main/scripts/metrics/Snakefile
--- scib-pipeline-main_orig/scripts/metrics/Snakefile	2022-10-14 19:37:05.000000000 +0200
+++ scib-pipeline-main/scripts/metrics/Snakefile	2023-02-07 16:15:17.272041060 +0100
@@ -17,7 +17,7 @@
         organism  = lambda wildcards: cfg.get_from_scenario(wildcards.scenario, key="organism"),
         assay     = lambda wildcards: cfg.get_from_scenario(wildcards.scenario, key="assay"),
         hvgs      = lambda wildcards: cfg.get_feature_selection(wildcards.hvg),
-        cmd       = f"conda run -n {cfg.py_env} python"
+        cmd       = f"python"
     shell:
         """
         {params.cmd} scripts/metrics/metrics.py -u {input.u} -i {input.i} \
@@ -53,7 +53,7 @@
         cfg.ROOT / "metrics.csv"
     message: "Merge all metrics"
     params:
-        cmd = f"conda run -n {cfg.py_env} python"
+        cmd = f"python"
     shell:
         """
         {params.cmd} scripts/metrics/merge_metrics.py -i {input.tables} \
diff -ruN scib-pipeline-main_orig/scripts/preprocessing/Snakefile scib-pipeline-main/scripts/preprocessing/Snakefile
--- scib-pipeline-main_orig/scripts/preprocessing/Snakefile	2022-10-14 19:37:05.000000000 +0200
+++ scib-pipeline-main/scripts/preprocessing/Snakefile	2023-02-07 16:15:35.216041978 +0100
@@ -15,13 +15,12 @@
         scale     = lambda wildcards: "-s" if wildcards.scaling == "scaled" else "",
         rout      = lambda wildcards: "-r" if wildcards.prep == "RDS" else "",
         seurat    = lambda wildcards: "-l" if wildcards.prep == "RDS" else "",
-        env       = f"conda run -n {cfg.py_env}"
     benchmark:
         str(cfg.ROOT / "{scenario}/prepare/{scaling}/{hvg}/prep_{prep}.benchmark")
         #join_path(cfg.get_filename_pattern("prepare", "directory_by_setting"), "prep_{prep}.benchmark")
     shell:
         """
-        {params.env} python scripts/preprocessing/runPP.py -i {input.adata} \
+        python scripts/preprocessing/runPP.py -i {input.adata} \
         -o {output} -b {params.batch_key} \
         --hvgs {params.hvgs} {params.scale} {params.rout} {params.seurat}
         """
diff -ruN scib-pipeline-main_orig/scripts/visualization/Snakefile scib-pipeline-main/scripts/visualization/Snakefile
--- scib-pipeline-main_orig/scripts/visualization/Snakefile	2022-10-14 19:37:05.000000000 +0200
+++ scib-pipeline-main/scripts/visualization/Snakefile	2023-02-07 16:15:48.588042661 +0100
@@ -19,7 +19,7 @@
     params:
         batch_key = lambda wildcards: cfg.get_from_scenario(wildcards.scenario, key="batch_key"),
         label_key = lambda wildcards: cfg.get_from_scenario(wildcards.scenario, key="label_key"),
-        cmd       = f"conda run -n {cfg.py_env} python"
+        cmd       = f"python"
     shell:
         """
         {params.cmd} scripts/visualization/save_embeddings.py \
diff -ruN scib-pipeline-main_orig/Snakefile scib-pipeline-main/Snakefile
--- scib-pipeline-main_orig/Snakefile	2022-10-14 19:37:05.000000000 +0200
+++ scib-pipeline-main/Snakefile	2023-02-07 16:13:38.536036013 +0100
@@ -32,5 +32,5 @@
         cfg.ROOT / "benchmarks.csv"
     message: "Merge all benchmarks"
     params:
-        cmd = f"conda run -n {cfg.py_env} python"
+        cmd = f"python"
     shell: "{params.cmd} {input.script} -o {output} --root {cfg.ROOT}"
