diff --git a/easybuild/framework/easyblock.py b/easybuild/framework/easyblock.py
index a515bc4c8..c0b7e3553 100644
--- a/easybuild/framework/easyblock.py
+++ b/easybuild/framework/easyblock.py
@@ -2378,6 +2378,14 @@ class EasyBlock(object):
                 raise EasyBuildError("EasyBuild-version %s is newer than the currently running one. Aborting!",
                                      easybuild_version)
 
+        source_deps = self.cfg['source_deps']
+        pre_fetch_env = None
+        # load modules for source dependencies (if any)
+        if source_deps:
+            pre_fetch_env = copy.deepcopy(os.environ)
+            source_deps_mod_names = [d['short_mod_name'] for d in source_deps]
+            self.modules_tool.load(source_deps_mod_names)
+
         start_progress_bar(PROGRESS_BAR_DOWNLOAD_ALL, self.cfg.count_files())
 
         if self.dry_run:
@@ -2464,6 +2472,9 @@ class EasyBlock(object):
 
         stop_progress_bar(PROGRESS_BAR_DOWNLOAD_ALL)
 
+        if pre_fetch_env:
+            restore_env(pre_fetch_env)
+
     def checksum_step(self):
         """Verify checksum of sources and patches, if a checksum is available."""
         for fil in self.src + self.patches:
diff --git a/easybuild/framework/easyconfig/default.py b/easybuild/framework/easyconfig/default.py
index 7c1b5acab..5ce8cdaca 100644
--- a/easybuild/framework/easyconfig/default.py
+++ b/easybuild/framework/easyconfig/default.py
@@ -170,6 +170,7 @@ DEFAULT_CONFIG = {
     'multi_deps_load_default': [True, "Load module for first version listed in multi_deps by default", DEPENDENCIES],
     'osdependencies': [[], "OS dependencies that should be present on the system", DEPENDENCIES],
     'moddependpaths': [None, "Absolute path(s) to prepend to MODULEPATH before loading dependencies", DEPENDENCIES],
+    'source_deps': [[], "source dependencies that should be present befory getting the sources", DEPENDENCIES],
 
     # LICENSE easyconfig parameters
     'accept_eula': [False, "Accepted End User License Agreement (EULA) for this software", LICENSE],
diff --git a/easybuild/framework/easyconfig/easyconfig.py b/easybuild/framework/easyconfig/easyconfig.py
index 57c910c74..04db92add 100644
--- a/easybuild/framework/easyconfig/easyconfig.py
+++ b/easybuild/framework/easyconfig/easyconfig.py
@@ -766,6 +766,14 @@ class EasyConfig(object):
                 builddeps = [self._parse_dependency(dep, build_only=True) for dep in builddeps]
             self['builddependencies'] = remove_false_versions(builddeps)
 
+            sourcedeps = self['source_deps']
+            if sourcedeps and all(isinstance(x, (list, tuple)) for b in sourcedeps for x in b):
+                self.iterate_options.append('source_deps')
+                sourcedeps = [[self._parse_dependency(dep, build_only=True) for dep in x] for x in sourcedeps]
+            else:
+                sourcedeps = [self._parse_dependency(dep, build_only=True) for dep in sourcedeps]
+            self['source_deps'] = remove_false_versions(sourcedeps)
+
             # keep track of parsed multi deps, they'll come in handy during sanity check & module steps...
             self.multi_deps = self.get_parsed_multi_deps()
 
