diff -ruN orig/include/c99_math.h fix/include/c99_math.h
--- orig/include/c99_math.h	2023-05-12 10:51:56.410611000 +0200
+++ fix/include/c99_math.h	2023-05-12 10:52:22.369775000 +0200
@@ -190,7 +190,7 @@
  * undefines those functions, which in glibc 2.23, are defined as macros rather
  * than functions as in glibc 2.22.
  */
-#if __cplusplus >= 201103L && (__GLIBC__ > 2 || (__GLIBC__ == 2 && __GLIBC_MINOR__ >= 23))
+#if __cplusplus >= 201103L && (__GLIBC__ > 2 || (__GLIBC__ == 2 && __GLIBC_MINOR__ >= 23 && __GLIBC_MINOR__ < 28 ))
 #include <cmath>
 
 using std::fpclassify;
diff -ruN orig/src/gallium/auxiliary/util/u_format_tests.c fix/src/gallium/auxiliary/util/u_format_tests.c
--- orig/src/gallium/auxiliary/util/u_format_tests.c	2023-05-12 12:06:46.907462000 +0200
+++ fix/src/gallium/auxiliary/util/u_format_tests.c	2023-05-12 12:06:26.110530000 +0200
@@ -919,7 +919,7 @@
    /* Max representable value */
    {PIPE_FORMAT_R16_FLOAT, PACKED_1x16(0xffff), PACKED_1x16(0x7bff), UNPACKED_1x1(    65504.0, 0.0, 0.0, 1.0)},
 
-#if !defined(PIPE_CC_MSVC)
+#if defined(PIPE_CC_MSVC)
 
    /* NaNs */
    {PIPE_FORMAT_R16_FLOAT, PACKED_1x16(0xffff), PACKED_1x16(0x7c01), UNPACKED_1x1(        NAN, 0.0, 0.0, 1.0)},
