--- vg/.gitmodules.orig 2023-05-10 08:25:37.689042043 +0200
+++ vg/.gitmodules      2023-05-10 08:26:14.299395085 +0200
@@ -63,7 +63,7 @@
        url = https://github.com/adamnovak/backward-cpp.git
 [submodule "deps/elfutils"]
        path = deps/elfutils
-       url = git://sourceware.org/git/elfutils.git
+       url = https://sourceware.org/git/elfutils.git
 [submodule "deps/structures"]
        path = deps/structures
        url = https://github.com/vgteam/structures.git
