--- AmberTools/test/Makefile.orig	2025-08-07 13:56:48.705382000 +0200
+++ AmberTools/test/Makefile	2025-08-07 14:00:13.526533000 +0200
@@ -74,7 +74,7 @@
 fast.unitcell:
 
 test.serial: clean is_amberhome_defined \
-	test.cpptraj test.pytraj test.antechamber test.mdgx \
+    test.antechamber test.mdgx \
 	test.leap test.unitcell test.resp test.py_resp test.reduce test.pbsa test.gbnsr6 \
 	test.parmed test.elsize test.sqm test.rism1d test.kmmd \
 	test.rism3d test.paramfit test.mm_pbsa \
@@ -84,7 +84,7 @@
 	test.Quick
 
 test.serial.nopython: clean is_amberhome_defined \
-	test.cpptraj test.antechamber test.mdgx \
+	test.antechamber test.mdgx \
 	test.leap test.unitcell test.resp test.reduce test.pbsa test.gbnsr6 \
 	test.elsize test.sqm test.rism1d test.kmmd \
 	test.rism3d test.paramfit test.mm_pbsa \
@@ -94,7 +94,7 @@
 	cd ../src/cpptraj/test && $(MAKE) -k test
 
 test.parallel2: clean is_amberhome_defined \
-	test.cpptraj test.rism3d test.bar_pbsa \
+	test.rism3d test.bar_pbsa \
 	test.Quick.MPI test.mmpbsa.parallel
 
 test.cuda.serial2: clean is_amberhome_defined \
@@ -141,8 +141,8 @@
 test.unitcell:
 	cd xtalutil/UnitCell && $(MAKE)
 
-test.cpptraj:
-	cd ../src/cpptraj/test && $(MAKE) -k test
+# test.cpptraj:
+# 	cd ../src/cpptraj/test && $(MAKE) -k test
 
 test.cpptraj.cuda:
 	cd ../src/cpptraj/test && $(MAKE) -e OPT=cuda -k test
@@ -150,8 +150,8 @@
 test.cpptraj.hip:
 	cd ../src/cpptraj/test && $(MAKE) -e OPT=hip -k test
 
-test.pytraj:
-	cd pytraj && $(MAKE)
+# test.pytraj:
+# 	cd pytraj && $(MAKE)
 
 test.pymsmt:
 	cd pymsmt/mcpb/g03-ff14SB && ./Run.pymsmt
@@ -414,7 +414,7 @@
 test.pysander:
 	cd ../../test/sanderapi && make Python
 
-test.python: test.pysander test.parmed test.pytraj test.pymsmt \
+test.python: test.pysander test.parmed test.pymsmt \
 	     test.cpinutil test.ceinutil test.cpeinutil test.genremdinputs test.fitpkaeo \
 	     test.fixremdcouts test.mmpbsa test.py_resp test.bar_pbsa
 
