--- bin/prop_furin.orig	2024-11-19 15:35:38.238817000 +0100
+++ bin/prop_furin	2024-11-22 11:25:25.893391000 +0100
@@ -1,4 +1,4 @@
-#! /bin/tcsh -f
+#! /bin/env -S tcsh -f
 
 
 ##########################################################################
@@ -24,6 +24,7 @@
 #	$PROPSYN/furin-ACD.syn
 #	$PROPSYN/furin-BCD.syn
 
+set verbose
 
 set howfile 	= $1			# 
 set number	= 1			# process only single sequence
