--- LISEM-6a5ef17d87bab4227d48673c95c12c61462114dd/CMakeLists.txt.orig	2022-08-10 22:06:03.000000000 +0200
+++ LISEM-6a5ef17d87bab4227d48673c95c12c61462114dd/CMakeLists.txt	2023-03-12 09:53:49.973922473 +0100
@@ -12,6 +12,8 @@
 
 SET(REMOVEFILESUNITTEST "")
 
+set(OpenMVS_LIBRARIES "-L$EBROOTOPENMVS/lib/OpenMVS -lMVS")
+
 
 # ==============================================================================
 # get the compilation settings
