--- LISEM-6a5ef17d87bab4227d48673c95c12c61462114dd/lisem/algorithms/sfm/Common/CUDA.cpp.orig	2022-08-10 22:06:03.000000000 +0200
+++ LISEM-6a5ef17d87bab4227d48673c95c12c61462114dd/lisem/algorithms/sfm/Common/CUDA.cpp	2023-03-16 18:25:42.711788870 +0100
@@ -6,10 +6,11 @@
 // (See http://www.boost.org/LICENSE_1_0.txt)
 
 #include "Common.h"
-#include "CUDA.h"
 
 #ifdef _USE_CUDA
 
+#include "CUDA.h"
+
 
 // D E F I N E S ///////////////////////////////////////////////////
 
