--- LISEM-6a5ef17d87bab4227d48673c95c12c61462114dd/lisem/algorithms/sfm/Common/Util.h.orig	2022-08-10 22:06:03.000000000 +0200
+++ LISEM-6a5ef17d87bab4227d48673c95c12c61462114dd/lisem/algorithms/sfm/Common/Util.h	2023-03-13 16:23:51.857794428 +0100
@@ -9,8 +9,6 @@
 #define __SEACAVE_UTIL_H__
 
 
-#define _MSC_VER 1950
-
 // I N C L U D E S /////////////////////////////////////////////////
 
 #ifdef _MSC_VER
