--- LISEM-6a5ef17d87bab4227d48673c95c12c61462114dd/lisem/model/seismic/setupRun.C.orig	2022-08-10 22:06:03.000000000 +0200
+++ LISEM-6a5ef17d87bab4227d48673c95c12c61462114dd/lisem/model/seismic/setupRun.C	2023-03-13 08:22:30.585399002 +0100
@@ -1666,7 +1666,7 @@
 // tmp
       if (mVerbose >=2) std::cout << "Calling mkdir() on path: " << pathsofar.str() << std::endl;
 // old code for recursively making the output directory
-       if (mkdir(pathsofar.str().c_str()) // why do we need group permissions?
+       if (mkdir(pathsofar.str().c_str(), S_IRWXU | S_IRWXG | S_IROTH | S_IXOTH) // why do we need group permissions?
           == -1)
       {
     if (mVerbose >=2) std::cout << "mkdir() returned an error code." << std::endl;
