--- gromacs-constantph/python_packaging/src/CMakeLists.txt.orig	2025-07-10 16:36:09.008342000 +0200
+++ gromacs-constantph/python_packaging/src/CMakeLists.txt	2025-07-10 16:39:01.928824000 +0200
@@ -116,7 +116,12 @@
     add_subdirectory(external/pybind)
 else()
     # Reference https://gitlab.com/gromacs/gromacs/-/issues/2896
-    find_package(pybind11 2.2 REQUIRED)
+    find_package(pybind11 2.11 REQUIRED)
+        if(NOT TARGET pybind11::headers)
+            add_library(pybind11::headers INTERFACE IMPORTED)
+            target_link_libraries(pybind11::headers INTERFACE pybind11::pybind11)
+            target_include_directories(pybind11::headers INTERFACE ${pybind11_INCLUDE_DIRS})
+        endif()
 endif()
 
 set(GMXAPI_PYTHON_EXTENSION_SOURCES
@@ -201,7 +206,7 @@
     # The Python module is being built for a GROMACS installation.
     set_target_properties(_gmxapi PROPERTIES BUILD_WITH_INSTALL_RPATH TRUE)
     set_target_properties(_gmxapi PROPERTIES INSTALL_RPATH_USE_LINK_PATH TRUE)
-    target_link_libraries(_gmxapi PRIVATE Gromacs::gmxapi)
+    target_link_libraries(_gmxapi PRIVATE pybind11::pybind11 Gromacs::gmxapi)
     # By default, scikit-build expects the library to be installed into a directory
     # named for the Python package as in setup.py.
     install(TARGETS _gmxapi LIBRARY DESTINATION gmxapi)
@@ -210,7 +215,7 @@
     # The Python module is being built against GROMACS in its build tree.
     # Note: we do not have plans to install the staged package when SKBUILD != TRUE
     set_target_properties(_gmxapi PROPERTIES BUILD_WITH_INSTALL_RPATH FALSE)
-    target_link_libraries(_gmxapi PRIVATE Gromacs::gmxapi)
+    target_link_libraries(_gmxapi PRIVATE pybind11::pybind11 Gromacs::gmxapi)
 
     # TODO: Determine packaging and installation cases and implementation.
     # Reference https://gitlab.com/gromacs/gromacs/-/issues/2896 for additional discussion.
