--- zip.h.orig	2025-07-01 18:19:39.457327000 +0200
+++ zip.h	2025-07-01 18:21:12.657206000 +0200
@@ -87,6 +87,8 @@
 /* Set up portability */
 #include "tailor.h"
 
+#include <string.h>
+
 #ifdef USE_ZLIB
 #  include "zlib.h"
 #endif
@@ -722,12 +724,6 @@
 
 int fcopy OF((FILE *, FILE *, uzoff_t));
 
-#ifdef ZMEM
-   char *memset OF((char *, int, unsigned int));
-   char *memcpy OF((char *, char *, unsigned int));
-   int memcmp OF((char *, char *, unsigned int));
-#endif /* ZMEM */
-
         /* in system dependent fileio code (<system>.c) */
 #ifndef UTIL
 # ifdef PROCNAME
