--- Source/JavaScriptCore/offlineasm/parser.rb.orig	2024-12-19 13:20:31.710087000 +0100
+++ Source/JavaScriptCore/offlineasm/parser.rb	2024-12-19 13:21:13.276547000 +0100
@@ -584,7 +584,8 @@
         firstCodeOrigin = @tokens[@idx].codeOrigin
         list = []
         loop {
-            if (@idx == @tokens.length and not final) or (final and @tokens[@idx] =~ final)
+            token_str = @tokens[@idx].is_a?(Annotation) ? @tokens[@idx].string : @tokens[@idx].to_s
+            if (@idx == @tokens.length && !final) || (final && token_str =~ final)
                 break
             elsif @tokens[@idx].is_a? Annotation
                 # This is the only place where we can encounter a global
@@ -790,7 +791,8 @@
         fileList = []
         fileList << @tokens[@idx].codeOrigin.fileName
         loop {
-            if (@idx == @tokens.length and not final) or (final and @tokens[@idx] =~ final)
+            token_str = @tokens[@idx].is_a?(Annotation) ? @tokens[@idx].string : @tokens[@idx].to_s
+            if (@idx == @tokens.length && !final) || (final && token_str =~ final)
                 break
             elsif @tokens[@idx] == "include"
                 @idx += 1
