put flexiblas on the first place to be used
--- CMake/FindCBLAS.cmake.orig	2024-03-15 18:09:25.303516149 +0100
+++ CMake/FindCBLAS.cmake	2024-03-15 18:12:38.716403155 +0100
@@ -11,13 +11,13 @@
 find_path(CBLAS_INCLUDE_DIRS NAMES cblas.h
   HINTS CBLAS_ROOT ENV CBLAS_ROOT
   PATHS ${INCLUDE_INSTALL_DIR} ${CMAKE_INSTALL_PREFIX}/include
-  PATH_SUFFIXES openblas cblas blis flexiblas
+  PATH_SUFFIXES flexiblas openblas cblas blis
   )
 
-find_library(CBLAS_LIBRARIES NAMES accelerate openblas cblas blas blis flexiblas
+find_library(CBLAS_LIBRARIES NAMES accelerate flexiblas openblas cblas blas blis
   HINTS CBLAS_ROOT ENV CBLAS_ROOT
   PATHS ${LIB_INSTALL_DIR} ${CMAKE_INSTALL_PREFIX}/lib
-  PATH_SUFFIXES openblas cblas blis flexiblas
+  PATH_SUFFIXES flexiblas openblas cblas blis
   )
 
 include(FindPackageHandleStandardArgs)
