diff -ruN MitoHiFi-3.2_orig/src/mitohifi.py MitoHiFi-3.2/src/mitohifi.py
--- MitoHiFi-3.2_orig/src/mitohifi.py	2023-11-18 18:02:10.410462654 +0000
+++ MitoHiFi-3.2/src/mitohifi.py	2023-11-18 18:13:04.180230454 +0000
@@ -53,7 +53,7 @@
     optional.add_argument("-p", help="-p: Percentage of query in the blast match with close-related mito", type=int, default=50, metavar='<PERC>')
     optional.add_argument("-m", help="-m: Number of bits for HiFiasm bloom filter [it maps to -f in HiFiasm] (default = 0)", type=int, default=0, metavar='<BLOOM FILTER>')
     optional.add_argument("--max-read-len", help="Maximum lenght of read relative to related mito (default = 1.0x related mito length)", type=float, default=1.0)
-    optional.add_argument("--mitos", help="Use MITOS2 for annotation (opposed to default MitoFinder", action="store_true")
+    optional.add_argument("--mitos", help="Not applicable in EasyBuild installation - MITOS2 the only possible option for annotation.", action="store_true")
     optional.add_argument('--circular-size', help='Size to consider when checking for circularization', type=int, default=220)
     optional.add_argument('--circular-offset', help='Offset from start and finish to consider when looking for circularization', type=int, default=40)
     optional.add_argument('-winSize', help='Size of windows to calculate coverage over the final_mitogenom', type=int, default=300)
@@ -70,6 +70,10 @@
         """, type=str, default='1', metavar='<GENETIC CODE>')
     args = parser.parse_args()
     
+    args.mitos = True
+    logging.info("MITOS2 is the only option for annotation in EasyBuild installation. Annotation will be done using MITOS2.")
+    
+    
     # Set log message format
     FORMAT='%(asctime)s [%(levelname)s] %(message)s'
 
