# Fix https://github.com/libxsmm/libxsmm/issues/339
--- include/libxsmm_macros.h.orig	2023-04-22 15:26:58.835286000 +0200
+++ include/libxsmm_macros.h	2023-04-22 15:36:33.650173000 +0200
@@ -536,14 +536,10 @@
 #endif
 
 /* _Float128 was introduced with GNU GCC 7.0. */
-#if !defined(_Float128) && defined(__GNUC__) && !defined(__cplusplus) \
-  && (LIBXSMM_VERSION3(7, 0, 0) > LIBXSMM_VERSION3(__GNUC__, __GNUC_MINOR__, __GNUC_PATCHLEVEL__) \
-  || (defined(__INTEL_COMPILER) && defined(__INTEL_COMPILER_UPDATE) && ( \
-        ((1800 <= ((__INTEL_COMPILER) + (__INTEL_COMPILER_UPDATE))) \
-      && (1801  > ((__INTEL_COMPILER) + (__INTEL_COMPILER_UPDATE)))) || \
-        ((1706  > ((__INTEL_COMPILER) + (__INTEL_COMPILER_UPDATE))) \
-      &&    (0 != ((__INTEL_COMPILER) + (__INTEL_COMPILER_UPDATE)))))))
-# define _Float128 __float128
+#if defined(__GLIBC__) && defined(__GLIBC_MINOR__) && LIBXSMM_VERSION2(__GLIBC__, __GLIBC_MINOR__) < LIBXSMM_VERSION2(2, 26)
+# if !defined(_Float128) && !defined(__SIZEOF_FLOAT128__) && defined(__GNUC__) && !defined(__cplusplus) && defined(__linux__)
+#   define _Float128 __float128
+# endif
 #endif
 
 #if defined(LIBXSMM_OFFLOAD_TARGET)
