--- defaultValues.py.orig	2024-04-22 16:32:59.951323371 +0200
+++ defaultValues.py	2024-04-22 16:34:47.872224925 +0200
@@ -28,7 +28,7 @@
 
     MODEL_DIVERGENCE_WARNING_THRESHOLD = 25
 
-    DB_LOCATION_DEFINITION = os.path.join(VERSION_PATH, 'diamond_path.json')
+    DB_LOCATION_DEFINITION = os.environ.get("CHECKM2DB", "Not Set")
     DB_VAR = "CHECKM2DB"
     try:
         DEFAULT_DB_INSTALL_LOCATION = os.path.join(str(Path.home()), 'databases')
