--- defaultValues.py.orig	2024-04-22 19:31:57.417114061 +0200
+++ defaultValues.py	2024-04-22 19:34:23.664324934 +0200
@@ -28,7 +28,7 @@
 
     MODEL_DIVERGENCE_WARNING_THRESHOLD = 25
 
-    DB_LOCATION_DEFINITION = os.path.join(VERSION_PATH, 'diamond_path.json')
+    DB_LOCATION_DEFINITION = os.eviron.get("CHECKM2DB", "Not Set")
     DB_VAR = "CHECKM2DB"
     try:
         DEFAULT_DB_INSTALL_LOCATION = os.path.join(str(Path.home()), 'databases')
