diff --git a/jax/_src/lax/pallas_lowerings/__init__.py b/jax/_src/lax/pallas_lowerings/__init__.py
new file mode 100644
--- /dev/null
+++ b/jax/_src/lax/pallas_lowerings/__init__.py
@@ -0,0 +1 @@
+"""Pallas lowerings for lax primitives."""
diff --git a/jax/_src/lax/pallas_lowerings/gpu/__init__.py b/jax/_src/lax/pallas_lowerings/gpu/__init__.py
new file mode 100644
--- /dev/null
+++ b/jax/_src/lax/pallas_lowerings/gpu/__init__.py
@@ -0,0 +1 @@
+"""GPU Pallas lowerings for lax primitives."""
