--- checkm2/defaultValues.py.orig	2024-04-22 16:32:59.951323371 +0200
+++ checkm2/defaultValues.py	2024-04-22 18:37:26.985171078 +0200
@@ -29,7 +29,7 @@
     MODEL_DIVERGENCE_WARNING_THRESHOLD = 25
 
     DB_LOCATION_DEFINITION = os.path.join(VERSION_PATH, 'diamond_path.json')
-    DB_VAR = "CHECKM2DB"
+    DB_VAR = "CHECKM3DB"
     try:
         DEFAULT_DB_INSTALL_LOCATION = os.path.join(str(Path.home()), 'databases')
     except:
