--- ni/src/ncl/h5data_struct.h.orig	2025-12-03 12:15:42.069056000 +0100
+++ ni/src/ncl/h5data_struct.h	2025-12-03 12:15:56.219997000 +0100
@@ -263,12 +263,24 @@
     string_list_t *tail;
 } string_queue_t;
 
-herr_t _NclHDF5search_obj(char *name, H5O_info_t *oinfo, void *_NclHDF5obj, char *already_seen);
-herr_t _NclHDF5search_link(char *name, H5O_info_t *oinfo, void *_NclHDF5link);
+herr_t _NclHDF5search_obj(const char *name,
+                          const H5O_info_t *oinfo,
+                          void *_NclHDF5obj,
+                          char *already_seen);
+
+herr_t _NclHDF5search_link(char *name,
+                           const H5L_info_t *linfo,
+                           void *_NclHDF5link);
 
 /* Typedefs for serach functions */
-typedef herr_t (*_NclHDF5search_obj_func_t) (char *path_name, H5O_info_t *oinfo, void *udata, char *already_seen);
-typedef herr_t (*_NclHDF5search_link_func_t) (char *path_name, H5L_info_t *linfo, void *udata);
+typedef herr_t (*_NclHDF5search_obj_func_t)(const char *name,
+                                            const H5O_info_t *oinfo,
+                                            void *_NclHDF5obj,
+                                            char *already_seen);
+
+typedef herr_t (*_NclHDF5search_link_func_t)(char *name,
+                                             const H5L_info_t *linfo,
+                                             void *_NclHDF5link);
 
 typedef struct
 {
@@ -301,7 +313,7 @@
 
 herr_t _NclHDF5dataset_info(hid_t dset, char *name, NclHDF5dataset_node_t *dataset_node);
 herr_t _NclHDF5dataset_attr(hid_t dset, char *name, NclHDF5dataset_node_t *dataset_node);
-herr_t _NclHDF5search_by_name(hid_t loc_id, char *path, H5L_info_t *linfo, void *_udata);
+herr_t _NclHDF5search_by_name(hid_t loc_id, const char *path, const H5L_info_t *linfo, void *_udata);
 
 void _NclHDF5_addr_add(NclHDF5_addr_t *visited, haddr_t addr, char *path);
 void setHDF5endian(hid_t type, hsize_t size, NclHDF5datatype_t *NclHDF5datatype);
@@ -310,8 +322,7 @@
 
 NclHDF5datatype_t *_NclHDF5get_typename(hid_t type, int ind);
 
-herr_t _NclHDF5check_attr(hid_t obj_id, char *attr_name, const H5A_info_t *ainfo,
-                          void *attr_data);
+herr_t _NclHDF5check_attr(hid_t obj_id, const char *attr_name, const H5A_info_t *ainfo, void *attr_data);
 
 void _NclHDF5print_group(NclHDF5group_node_t *HDF5group);
 void _NclHDF5free_group(NclHDF5group_node_t *HDF5group);
