--- by_key.rs.orig	2025-04-28 10:15:53.938275000 +0200
+++ by_key.rs	2025-04-28 13:13:42.696591000 +0200
@@ -202,76 +202,84 @@
 
             // Wrap this in a closure so that a failure to send (which signifies a failure) can be
             // caught while waiting for tasks.
-            let mut receive_and_pass = async || {
-                while let Ok(mut lin_rx) = io_rx.recv().await {
-                    while let Some(Priority(Reverse(seq), (source_token, partitions))) =
-                        lin_rx.get().await
-                    {
-                        for (row_encoded, keys, partition) in partitions {
-                            let num_open_partitions = open_partitions.len();
-                            let open_partition = match open_partitions.get_mut(&row_encoded) {
-                                None if num_open_partitions >= max_open_partitions => {
-                                    if num_open_partitions == max_open_partitions && verbose {
-                                        eprintln!(
-                                            "[partition[by-key]]: Reached maximum open partitions. Buffering the rest to memory before writing.",
+            let receive_and_pass = {
+                let base_path = base_path.clone();
+                let file_path_cb = file_path_cb.clone();
+                let create_new_sink = create_new_sink.clone();
+                let sink_input_schema = sink_input_schema.clone();
+                let ext = ext.clone();
+                let state = state.clone();
+                
+                async move {
+                    while let Ok(mut lin_rx) = io_rx.recv().await {
+                        while let Some(Priority(Reverse(seq), (source_token, partitions))) = lin_rx.get().await {
+                            for (row_encoded, keys, partition) in partitions {
+                                let num_open_partitions = open_partitions.len();
+                                let open_partition = match open_partitions.get_mut(&row_encoded) {
+                                    None if num_open_partitions >= max_open_partitions => {
+                                        if num_open_partitions == max_open_partitions && verbose {
+                                            eprintln!(
+                                                "[partition[by-key]]: Reached maximum open partitions. Buffering the rest to memory before writing.",
+                                            );
+                                        }
+            
+                                        let (idx, previous) = open_partitions.insert_full(
+                                            row_encoded,
+                                            OpenPartition::Buffer(keys, Vec::new()),
                                         );
-                                    }
-
-                                    let (idx, previous) = open_partitions.insert_full(
-                                        row_encoded,
-                                        OpenPartition::Buffer(keys, Vec::new()),
-                                    );
-                                    debug_assert!(previous.is_none());
-                                    open_partitions.get_index_mut(idx).unwrap().1
-                                },
-                                None => {
-                                    let result = open_new_sink(
-                                        base_path.as_path(),
-                                        file_path_cb.as_ref(),
-                                        super::default_by_key_file_path_cb,
-                                        file_idx,
-                                        file_idx,
-                                        0,
-                                        Some(keys.as_slice()),
-                                        &create_new_sink,
-                                        sink_input_schema.clone(),
-                                        "by-key",
-                                        ext.as_str(),
-                                        verbose,
-                                        &state,
-                                    ).await?;
-                                    file_idx += 1;
-
-                                    let Some((join_handles, sender)) = result else {
-                                        return Ok(());
-                                    };
-
-                                    let (idx, previous) = open_partitions.insert_full(
-                                        row_encoded,
-                                        OpenPartition::Sink(sender, join_handles),
-                                    );
-                                    debug_assert!(previous.is_none());
-                                    open_partitions.get_index_mut(idx).unwrap().1
-                                },
-                                Some(open_partition) => open_partition,
-                            };
-
-                            match open_partition {
-                                OpenPartition::Sink(input, _) => {
-                                    let morsel = Morsel::new(partition, seq, source_token.clone());
-                                    if input.send(morsel).await.is_err() {
-                                        return Ok(());
-                                    }
-                                },
-                                OpenPartition::Buffer(_keys, buffer) => buffer.push(partition),
+                                        debug_assert!(previous.is_none());
+                                        open_partitions.get_index_mut(idx).unwrap().1
+                                    },
+                                    None => {
+                                        let result = open_new_sink(
+                                            base_path.as_path(),
+                                            file_path_cb.as_ref(),
+                                            super::default_by_key_file_path_cb,
+                                            file_idx,
+                                            file_idx,
+                                            0,
+                                            Some(keys.as_slice()),
+                                            &create_new_sink,
+                                            sink_input_schema.clone(),
+                                            "by-key",
+                                            ext.as_str(),
+                                            verbose,
+                                            &state,
+                                        ).await?;
+                                        file_idx += 1;
+            
+                                        let Some((join_handles, sender)) = result else {
+                                            return Ok(());
+                                        };
+            
+                                        let (idx, previous) = open_partitions.insert_full(
+                                            row_encoded,
+                                            OpenPartition::Sink(sender, join_handles),
+                                        );
+                                        debug_assert!(previous.is_none());
+                                        open_partitions.get_index_mut(idx).unwrap().1
+                                    },
+                                    Some(open_partition) => open_partition,
+                                };
+            
+                                match open_partition {
+                                    OpenPartition::Sink(input, _) => {
+                                        let morsel = Morsel::new(partition, seq, source_token.clone());
+                                        if input.send(morsel).await.is_err() {
+                                            return Ok(());
+                                        }
+                                    },
+                                    OpenPartition::Buffer(_keys, buffer) => buffer.push(partition),
+                                }
                             }
                         }
                     }
+            
+                    PolarsResult::Ok(())
                 }
-
-                PolarsResult::Ok(())
             };
-            receive_and_pass().await?;
+            
+            receive_and_pass.await?;
 
             // At this point, we need to wait for all sinks to finish writing and close them. Also,
             // sinks that ended up buffering need to output their data.
