--- crates/polars-stream/src/nodes/io_sinks/partition/by_key.rs.orig	2025-04-24 15:25:10.518214000 +0200
+++ crates/polars-stream/src/nodes/io_sinks/partition/by_key.rs	2025-04-24 15:27:42.591896000 +0200
@@ -202,7 +202,7 @@
 
             // Wrap this in a closure so that a failure to send (which signifies a failure) can be
             // caught while waiting for tasks.
-            let mut receive_and_pass = async || {
+            let mut receive_and_pass = async {
                 while let Ok(mut lin_rx) = io_rx.recv().await {
                     while let Some(Priority(Reverse(seq), (source_token, partitions))) =
                         lin_rx.get().await
@@ -271,7 +271,7 @@
 
                 PolarsResult::Ok(())
             };
-            receive_and_pass().await?;
+            receive_and_pass.await?;
 
             // At this point, we need to wait for all sinks to finish writing and close them. Also,
             // sinks that ended up buffering need to output their data.
