--- bootstrap_orig.py	2022-11-02 14:59:18.639944541 +0100
+++ bootstrap.py	2022-11-02 14:59:39.698292000 +0100
@@ -1872,7 +1872,7 @@
 
     configcmd =[
         self.python_base, # default to using our python rather than system python
-        self.opjoin('..', 'modules', 'cctbx_project', 'libtbx', 'configure.py')
+        self.opjoin('modules', 'cctbx_project', 'libtbx', 'configure.py')
         ] + self.get_libtbx_configure() + self.config_flags
     self.add_step(self.shell(command=configcmd, workdir=[_BUILD_DIR],
       description="run configure.py", env=env))
