--- cctbx-2023.5.orig/lib/libtbx/auto_build/install_distribution.py	2023-06-01 18:41:00.000000000 +0200
+++ cctbx-2023.5/lib/libtbx/auto_build/install_distribution.py	2023-06-24 09:45:55.865812651 +0200
@@ -341,7 +341,8 @@
       os.makedirs(self.tmp_dir)
     if (sys.platform != "darwin"):
       os.environ["LD_LIBRARY_PATH"] = "lib:%s:%s" % \
-        (op.join(self.base_dir, "lib"), op.join(self.base_dir, "lib64"))
+        (op.join(self.base_dir, "lib"), op.join(self.base_dir, "lib64"), os.getenv("LD_LIBRARY_PATH"))
+      print("LD_LIBRARY_PATH=%s" % os.getenv("LD_LIBRARY_PATH"))
     self.product_specific_binary_install(log=log)
 
     os.chdir(self.build_dir)
@@ -448,7 +449,8 @@
     if (self.flag_build_gui) and (sys.platform != "darwin") and \
       (not self.base_dir.endswith('conda_base')):
       os.environ["LD_LIBRARY_PATH"] = "lib:%s:%s" % \
-        (op.join(self.base_dir, "lib"), op.join(self.base_dir, "lib64"))
+        (op.join(self.base_dir, "lib"), op.join(self.base_dir, "lib64"), os.getenv("LD_LIBRARY_PATH"))
+      print("LD_LIBRARY_PATH=%s" % os.getenv("LD_LIBRARY_PATH"))
       regenerate_module_files.run(
         os.path.join(self.dest_dir, 'base'),
         out=self.out)
--- cctbx-2023.5.orig/modules/cctbx_project/libtbx/auto_build/install_distribution.py	2023-06-01 18:41:00.000000000 +0200
+++ cctbx-2023.5/modules/cctbx_project/libtbx/auto_build/install_distribution.py	2023-06-24 09:45:40.254251268 +0200
@@ -341,7 +341,8 @@
       os.makedirs(self.tmp_dir)
     if (sys.platform != "darwin"):
       os.environ["LD_LIBRARY_PATH"] = "lib:%s:%s" % \
-        (op.join(self.base_dir, "lib"), op.join(self.base_dir, "lib64"))
+        (op.join(self.base_dir, "lib"), op.join(self.base_dir, "lib64"), os.getenv("LD_LIBRARY_PATH"))
+      print("LD_LIBRARY_PATH=%s" % os.getenv("LD_LIBRARY_PATH"))
     self.product_specific_binary_install(log=log)
 
     os.chdir(self.build_dir)
@@ -448,7 +449,8 @@
     if (self.flag_build_gui) and (sys.platform != "darwin") and \
       (not self.base_dir.endswith('conda_base')):
       os.environ["LD_LIBRARY_PATH"] = "lib:%s:%s" % \
-        (op.join(self.base_dir, "lib"), op.join(self.base_dir, "lib64"))
+        (op.join(self.base_dir, "lib"), op.join(self.base_dir, "lib64"), os.getenv("LD_LIBRARY_PATH"))
+      print("LD_LIBRARY_PATH=%s" % os.getenv("LD_LIBRARY_PATH"))
       regenerate_module_files.run(
         os.path.join(self.dest_dir, 'base'),
         out=self.out)
--- cctbx-2023.5.orig/modules/cctbx_project/boost_adaptbx/boost/python.py	2023-06-01 18:40:59.000000000 +0200
+++ cctbx-2023.5/modules/cctbx_project/boost_adaptbx/boost/python.py	2023-06-24 11:12:26.163950598 +0200
@@ -37,6 +37,7 @@
     sys.setdlopenflags(previous_dlopenflags)
   return mod
 
+print("LD_LIBRARY_PATH=%s" % os.getenv('LD_LIBRARY_PATH'))
 ext = import_ext("boost_python_meta_ext")
 
 try: streambuf = ext.streambuf
