--- a/gems/itkMGHImageIO.h
+++ b/gems/itkMGHImageIO.h
@@ -113,24 +113,24 @@
     // type
     switch( m_ComponentType )
       {
-      case UCHAR:
+      case CommonEnums::IOComponent::UCHAR:
         writer.Write( fs::MRI_UCHAR ); break;
-      case USHORT:
+      case CommonEnums::IOComponent::USHORT:
         itkWarningMacro( << " Casting from USHORT to INT!!!" );
-      case INT:
+      case CommonEnums::IOComponent::INT:
         writer.Write( fs::MRI_INT ); break;
-      case DOUBLE:
+      case CommonEnums::IOComponent::DOUBLE:
         itkWarningMacro( << " Casting from DOUBLE to INT!!!" );
-      case FLOAT:
+      case CommonEnums::IOComponent::FLOAT:
         writer.Write( fs::MRI_FLOAT ); break;
-      case SHORT:
+      case CommonEnums::IOComponent::SHORT:
         writer.Write( fs::MRI_SHORT ); break;

       // DJ -- added these cases to make the compiler shut up
-      case UNKNOWNCOMPONENTTYPE:
-      case CHAR:
-      case UINT:
-      case ULONG:
-      case LONG:
+      case CommonEnums::IOComponent::UNKNOWNCOMPONENTTYPE:
+      case CommonEnums::IOComponent::CHAR:
+      case CommonEnums::IOComponent::UINT:
+      case CommonEnums::IOComponent::ULONG:
+      case CommonEnums::IOComponent::LONG:
         break;

       }
--- a/gems/itkMGHImageIO.cxx
+++ b/gems/itkMGHImageIO.cxx
@@ -142,9 +142,9 @@
     m_Dimensions[2] = uzero;

     if ( ByteSwapper<int>::SystemIsBigEndian())
-      m_ByteOrder = BigEndian;
+      m_ByteOrder = CommonEnums::IOByteOrder::BigEndian;
     else
-      m_ByteOrder = LittleEndian;
+      m_ByteOrder = CommonEnums::IOByteOrder::LittleEndian;

   }

@@ -251,16 +251,16 @@
     // Convert type to an ITK type
     switch( type )
       {
-      case fs::MRI_UCHAR: 
-	m_ComponentType = UCHAR; break;
+      case fs::MRI_UCHAR: 
+	m_ComponentType = CommonEnums::IOComponent::UCHAR; break;
       case fs::MRI_INT:
-	m_ComponentType = INT; break;
+	m_ComponentType = CommonEnums::IOComponent::INT; break;
       case fs::MRI_FLOAT:
-	m_ComponentType = FLOAT; break;
+	m_ComponentType = CommonEnums::IOComponent::FLOAT; break;
       case fs::MRI_SHORT:
-	m_ComponentType = SHORT; break;
+	m_ComponentType = CommonEnums::IOComponent::SHORT; break;
       case fs::MRI_TENSOR:
-	m_ComponentType = FLOAT; m_NumberOfComponents = 9; break;
+	m_ComponentType = CommonEnums::IOComponent::FLOAT; m_NumberOfComponents = 9; break;
       default:
 	itkExceptionMacro(<<" Unknown data type " << type << " using float by default.");
-	m_ComponentType = FLOAT;
+	m_ComponentType = CommonEnums::IOComponent::FLOAT;
       }

@@ -471,16 +471,16 @@
     switch(m_ComponentType)
       {
-      case UCHAR:
+      case CommonEnums::IOComponent::UCHAR:
 	ByteSwapper<unsigned char>::SwapRangeFromSystemToBigEndian((unsigned char*)buffer,
 								   numberOfPixels);
 	break;
-      case SHORT:
+      case CommonEnums::IOComponent::SHORT:
 	ByteSwapper<short>::SwapRangeFromSystemToBigEndian((short*)buffer,
 							   numberOfPixels);
 	break;
-      case INT:
+      case CommonEnums::IOComponent::INT:
 	ByteSwapper<int>::SwapRangeFromSystemToBigEndian((int*)buffer,
 							 numberOfPixels);
 	break;
-      case FLOAT:
+      case CommonEnums::IOComponent::FLOAT:
 	ByteSwapper<float>::SwapRangeFromSystemToBigEndian((float*)buffer,
 							   numberOfPixels);
 	break;
@@ -607,19 +607,19 @@
     unsigned int returnValue = 0;
     switch ( m_ComponentType )
       {
-      case UCHAR: returnValue = sizeof(unsigned char); break;
-      case SHORT: returnValue = sizeof(short); break;
-      case INT:   returnValue = sizeof(int); break;
-      case FLOAT: returnValue = sizeof(float); break;
+      case CommonEnums::IOComponent::UCHAR: returnValue = sizeof(unsigned char); break;
+      case CommonEnums::IOComponent::SHORT: returnValue = sizeof(short); break;
+      case CommonEnums::IOComponent::INT:   returnValue = sizeof(int); break;
+      case CommonEnums::IOComponent::FLOAT: returnValue = sizeof(float); break;

       // DJ -- added this in to get the compiler to shut up
-      case UNKNOWNCOMPONENTTYPE:
-      case CHAR:
-      case USHORT:
-      case UINT:
-      case ULONG:
-      case LONG:
-      case DOUBLE:
+      case CommonEnums::IOComponent::UNKNOWNCOMPONENTTYPE:
+      case CommonEnums::IOComponent::CHAR:
+      case CommonEnums::IOComponent::USHORT:
+      case CommonEnums::IOComponent::UINT:
+      case CommonEnums::IOComponent::ULONG:
+      case CommonEnums::IOComponent::LONG:
+      case CommonEnums::IOComponent::DOUBLE:
         break;
       }
     return returnValue;
