--- test_streams.py.orig	2026-05-27 14:11:57.291604000 +0200
+++ test_streams.py	2026-05-27 14:12:54.369398000 +0200
@@ -197,9 +197,7 @@
         stream.seek(1024)
         assert_(stream.all_data_read())
 
-    @pytest.mark.skipif(
-            (platform.system() == 'Windows' and sys.version_info >= (3, 14)),
-            reason='gh-23185')
+    @pytest.mark.skipif(sys.version_info >= (3, 14), reason='gh-23185')
     def test_all_data_read_overlap(self):
         COMPRESSION_LEVEL = 6
 
@@ -216,9 +214,7 @@
         stream.seek(len(data))
         assert_(stream.all_data_read())
 
-    @pytest.mark.skipif(
-            (platform.system() == 'Windows' and sys.version_info >= (3, 14)),
-            reason='gh-23185')
+    @pytest.mark.skipif(sys.version_info >= (3, 14), reason='gh-23185')
     def test_all_data_read_bad_checksum(self):
         COMPRESSION_LEVEL = 6
 
