--- cmake/hictkConfig.cmake.in.orig	2025-04-09 11:28:30.000000000 +0200
+++ cmake/hictkConfig.cmake.in	2025-05-20 08:51:31.976853246 +0200
@@ -13,7 +13,18 @@
     find_dependency(Arrow CONFIG QUIET REQUIRED)
 endif()
 
-find_dependency(bshoshany-thread-pool CONFIG QUIET REQUIRED)
+include(FetchContent)
+set(FETCHCONTENT_UPDATES_DISCONNECTED ON)
+FetchContent_Declare(
+    bshoshany_thread_pool
+    GIT_REPOSITORY https://github.com/bshoshany/thread-pool.git
+    GIT_TAG v5.0.0
+    DOWNLOAD_EXTRACT_TIMESTAMP TRUE
+    EXCLUDE_FROM_ALL
+    SYSTEM
+)
+FetchContent_MakeAvailable(bshoshany_thread_pool)
+
 find_dependency(concurrentqueue CONFIG QUIET REQUIRED)
 
 if(HICTK_WITH_EIGEN)
--- src/libhictk/balancing/CMakeLists.txt.orig  2025-04-09 11:28:30.000000000 +0200
+++ src/libhictk/balancing/CMakeLists.txt       2025-05-20 09:15:06.926899949 +0200
@@ -2,8 +2,25 @@
 #
 # SPDX-License-Identifier: MIT
 
-find_package(bshoshany-thread-pool REQUIRED QUIET)
-find_package(phmap REQUIRED QUIET)
+include(FetchContent)
+set(FETCHCONTENT_UPDATES_DISCONNECTED ON)
+FetchContent_Declare(
+    bshoshany_thread_pool
+    GIT_REPOSITORY https://github.com/bshoshany/thread-pool.git
+    GIT_TAG v5.0.0
+    DOWNLOAD_EXTRACT_TIMESTAMP TRUE
+    EXCLUDE_FROM_ALL
+    SYSTEM
+)
+FetchContent_MakeAvailable(bshoshany_thread_pool)
+
+FetchContent_Declare(
+    parallel-hashmap
+    GIT_REPOSITORY https://github.com/greg7mdp/parallel-hashmap.git
+    GIT_TAG        v2.0.0    # adjust tag/branch/commit as needed
+)
+FetchContent_MakeAvailable(parallel-hashmap)
+
 find_package(span-lite REQUIRED QUIET)
 find_package(zstd REQUIRED QUIET)
 
@@ -50,7 +67,7 @@
   INTERFACE
     bshoshany-thread-pool::bshoshany-thread-pool
     nonstd::span-lite
-    phmap
+    parallel-hashmap
     "zstd::libzstd_$<IF:$<BOOL:${BUILD_SHARED_LIBS}>,shared,static>"
 )
 
--- src/libhictk/hic/CMakeLists.txt.orig
+++ src/libhictk/hic/CMakeLists.txt
@@ -2,13 +2,31 @@
 #
 # SPDX-License-Identifier: MIT

-find_package(bshoshany-thread-pool REQUIRED QUIET)
+include(FetchContent)
+set(FETCHCONTENT_UPDATES_DISCONNECTED ON)
+FetchContent_Declare(
+    bshoshany_thread_pool
+    GIT_REPOSITORY https://github.com/bshoshany/thread-pool.git
+    GIT_TAG v5.0.0
+    DOWNLOAD_EXTRACT_TIMESTAMP TRUE
+    EXCLUDE_FROM_ALL
+    SYSTEM
+)
+FetchContent_MakeAvailable(bshoshany_thread_pool)
 find_package(concurrentqueue REQUIRED QUIET)
 find_package(fmt REQUIRED QUIET)
 find_package(libdeflate REQUIRED QUIET)
-find_package(phmap REQUIRED QUIET)
+FetchContent_MakeAvailable(bshoshany_thread_pool)
+
+FetchContent_Declare(
+    parallel-hashmap
+    GIT_REPOSITORY https://github.com/greg7mdp/parallel-hashmap.git
+    GIT_TAG        v2.0.0    # adjust tag/branch/commit as needed
+)
+FetchContent_MakeAvailable(parallel-hashmap)
+
 find_package(readerwriterqueue REQUIRED QUIET)
-find_package(zstd REQUIRED QUIET)
+find_library(zstd PATHS $ENV{EBROOTZSTD}/lib64)

 set(HICTK_CXX_BYTE_ORDER "${CMAKE_CXX_BYTE_ORDER}" CACHE STRING "Specify the CPU architecture endianness")

@@ -105,7 +123,7 @@ target_link_system_libraries(
     concurrentqueue::concurrentqueue
     fmt::fmt-header-only
     "libdeflate::libdeflate_$<IF:$<BOOL:${BUILD_SHARED_LIBS}>,shared,static>"
-    phmap
+    parallel-hashmap
     readerwriterqueue::readerwriterqueue
     "zstd::libzstd_$<IF:$<BOOL:${BUILD_SHARED_LIBS}>,shared,static>"
 )
--- src/libhictk/reference/CMakeLists.txt.orig
+++ src/libhictk/reference/CMakeLists.txt
@@ -2,8 +2,6 @@
 #
 # SPDX-License-Identifier: MIT

-find_package(phmap REQUIRED QUIET)
-
 add_library(hictk_internal_reference INTERFACE)
 add_library(hictk::reference ALIAS hictk_internal_reference)

--- src/hictk/load/CMakeLists.txt.orig  2025-04-09 20:28:31.000000000 +0200
+++ src/hictk/load/CMakeLists.txt       2025-08-12 12:57:47.348050079 +0200
@@ -2,9 +2,26 @@
 #
 # SPDX-License-Identifier: MIT
 
-find_package(bshoshany-thread-pool REQUIRED QUIET)
+include(FetchContent)
+set(FETCHCONTENT_UPDATES_DISCONNECTED ON)
+FetchContent_Declare(
+    bshoshany_thread_pool
+    GIT_REPOSITORY https://github.com/bshoshany/thread-pool.git
+    GIT_TAG v5.0.0
+    DOWNLOAD_EXTRACT_TIMESTAMP TRUE
+    EXCLUDE_FROM_ALL
+    SYSTEM
+)
+FetchContent_MakeAvailable(bshoshany_thread_pool)
+
+FetchContent_Declare(
+    parallel-hashmap
+    GIT_REPOSITORY https://github.com/greg7mdp/parallel-hashmap.git
+    GIT_TAG        v2.0.0    # adjust tag/branch/commit as needed
+)
+FetchContent_MakeAvailable(parallel-hashmap)
+
 find_package(fmt REQUIRED QUIET)
-find_package(phmap REQUIRED QUIET)
 find_package(readerwriterqueue REQUIRED QUIET)
 find_package(spdlog REQUIRED QUIET)
 find_package(Filesystem REQUIRED QUIET)
@@ -60,7 +77,7 @@
     readerwriterqueue::readerwriterqueue
     spdlog::spdlog_header_only
     std::filesystem
-    phmap
+    parallel-hashmap
   PUBLIC
     Threads::Threads
 )

