--- setup.py.orig	2024-02-19 11:57:46.377990148 +0100
+++ setup.py	2024-02-19 12:28:32.167593635 +0100
@@ -23,6 +23,20 @@
         'easymocap.annotator',
         'easymocap.estimator'
     ],
-    install_requires=[],
+    install_requires=[
+        'torch',
+        'torchvision',
+        'ipdb',
+        'joblib',
+        'tqdm',
+        'opencv-contrib-python',
+        'yacs',
+        'tabulate',
+        'termcolor',
+        'chumpy',
+        'mediapipe',
+        'func_timeout',
+        'pyrender',
+    ],
     data_files = []
 )
