--- scine_autocas/tests/test_main_functions.py.orig	2025-06-25 10:29:10.730537000 +0200
+++ scine_autocas/tests/test_main_functions.py	2025-06-25 10:29:43.328486000 +0200
@@ -21,27 +21,27 @@
         self.dummy_project_dir = self.path + "/dummy_dir"
         os.makedirs(self.dummy_project_dir, exist_ok=True)
 
-    def tearDown(self):
-        try:
-            shutil.rmtree(self.dummy_project_dir)
-        except FileNotFoundError:
-            pass
-        try:
-            shutil.rmtree(self.path + "/autocas_project")
-        except FileNotFoundError:
-            pass
-        try:
-            shutil.rmtree(self.path + "/../../autocas_project")
-        except FileNotFoundError:
-            pass
-        try:
-            shutil.rmtree(self.path + "/molcas_scratch")
-        except FileNotFoundError:
-            pass
-        try:
-            shutil.rmtree(self.path + "/../../molcas_scratch")
-        except FileNotFoundError:
-            pass
+    # def tearDown(self):
+    #     try:
+    #         shutil.rmtree(self.dummy_project_dir)
+    #     except FileNotFoundError:
+    #         pass
+    #     try:
+    #         shutil.rmtree(self.path + "/autocas_project")
+    #     except FileNotFoundError:
+    #         pass
+    #     try:
+    #         shutil.rmtree(self.path + "/../../autocas_project")
+    #     except FileNotFoundError:
+    #         pass
+    #     try:
+    #         shutil.rmtree(self.path + "/molcas_scratch")
+    #     except FileNotFoundError:
+    #         pass
+    #     try:
+    #         shutil.rmtree(self.path + "/../../molcas_scratch")
+    #     except FileNotFoundError:
+    #         pass
 
     def test_conventional(self):
         mainfunctions = MainFunctions()
