scine_autocas/interfaces/molcas/__init__.py
--- scine_autocas/interfaces/molcas/__init__.py.orig	2025-06-06 12:17:35.031662000 +0200
+++ scine_autocas/interfaces/molcas/__init__.py	2025-06-06 12:18:21.299490000 +0200
@@ -617,6 +617,9 @@
                     s1_list.append(self.qc_maquis.s1_entropy)
                     s2_list.append(self.qc_maquis.s2_entropy)
                     mutual_information_list.append(self.qc_maquis.mutual_information)
+                print("s1_list:", hasattr(self, "s1_list"))
+                print("s2_list:", hasattr(self, "s2_list"))
+                print("mutual_information_list:", hasattr(self, "mutual_information_list"))
                 self.s1_list = s1_list
                 self.s2_list = s2_list
                 self.mutual_information_list = mutual_information_list
