scine_autocas/interfaces/molcas/__init__.py
--- scine_autocas/interfaces/molcas/__init__.py.orig	2025-06-06 12:17:35.031662000 +0200
+++ scine_autocas/interfaces/molcas/__init__.py	2025-06-06 12:18:21.299490000 +0200
@@ -599,11 +599,18 @@
 
     def make_dmrg_measurements(self):
         """Build all entropies from QCMaquis HDF5 file."""
+        self.s1_list = None
+        self.s2_list = None
+        self.mutual_information_list = None
         if self.settings.n_excited_states > 0:
+            print(f"DDDD - N_EX_STATES: {self.settings.n_excited_states}")
+            print("Looking for:", self.settings.work_dir + "/" + self.project_name + ".results_state.0.h5")
+            print("Dir contents:", os.listdir(self.settings.work_dir))
             s1_list = []
             s2_list = []
             mutual_information_list = []
             if os.path.isfile(self.settings.work_dir + "/" + self.project_name + ".results_state.0.h5"):
+                print(f"EEEEE - RESULT EXISTS")
                 for i in range(self.settings.n_excited_states):
                     self.qc_maquis.read_hdf5(
                         self.settings.work_dir
@@ -614,9 +621,20 @@
                         + ".h5"
                     )
                     self.qc_maquis.make_diagnostics()
+                    print("BBBBBB QC_MAQUIS_ENTROPY: ")
+                    print(f"ENTROPY1: {self.qc_maquis.s1_entropy}")
+                    type_s1_entropy = type(self.qc_maquis.s1_entropy)
+                    print(f"ENTROPY1_TYPE: {type_s1_entropy}")
+                    print(f"ENTROPY2: {self.qc_maquis.s2_entropy}")
+                    type_s2_entropy = type(self.qc_maquis.s2_entropy)
+                    print(f"ENTROPY2_TYPE: {type_s2_entropy}")
                     s1_list.append(self.qc_maquis.s1_entropy)
                     s2_list.append(self.qc_maquis.s2_entropy)
                     mutual_information_list.append(self.qc_maquis.mutual_information)
+                print('CCCCCCC: ')
+                print(s1_list)
+                print(s2_list)
+                print(mutual_information_list)
                 self.s1_list = s1_list
                 self.s2_list = s2_list
                 self.mutual_information_list = mutual_information_list
@@ -625,17 +643,22 @@
                 pass
         else:
             if os.path.isfile(self.settings.work_dir + "/" + self.project_name + ".results_state.0.h5"):
+                print("FFFFF")
                 self.qc_maquis.read_hdf5(self.settings.work_dir + "/" + self.project_name + ".results_state.0.h5")
                 self.qc_maquis.make_diagnostics()
             else:
                 # no dmrg
                 pass
         if os.path.isfile(self.settings.work_dir + "/" + self.project_name + ".dmrgscf.h5"):
+            print("GGGGGGG")
             self.hdf5_utils.get_energy(self.settings.work_dir + "/" + self.project_name + ".dmrgscf.h5")
         elif os.path.isfile(self.settings.work_dir + "/" + self.project_name + ".rasscf.h5"):
+            print("HHHHHHHHH")
             self.hdf5_utils.get_energy(self.settings.work_dir + "/" + self.project_name + ".rasscf.h5")
         else:
+            print("IIIIIIIIII")
             pass
+        print("JJJJJJJJJJJJ")
         self.get_caspt2_from_log(self.settings.work_dir + "/" + self.project_name + ".log")
 
     def _convert_type_indices(self, type_indices):
