diff -ruN open_SURFEX_V8_1_orig/src/configure open_SURFEX_V8_1/src/configure
--- open_SURFEX_V8_1_orig/src/configure	2023-03-08 15:41:53.162714819 +0100
+++ open_SURFEX_V8_1/src/configure	2023-03-09 17:06:29.484464000 +0100
@@ -47,7 +47,7 @@
     export MVWORK=${MVWORK:-NO}
     export MNHENV=${MNHENV:-"
 #start new subshell
-bash --norc
+bash
 #sh
 "}
   ;;
@@ -64,7 +64,7 @@
     export MVWORK=${MVWORK:-NO}
     export MNHENV=${MNHENV:-"
 #start new subshell
-bash --norc
+bash
 #sh
 "}
   ;;
@@ -81,7 +81,7 @@
     export MVWORK=${MVWORK:-NO}
     export MNHENV=${MNHENV:-"
 #start new subshell
-bash --norc
+bash
 #sh
 "}
   ;;
diff -ruN open_SURFEX_V8_1_orig/src/Rules.zifort.mk open_SURFEX_V8_1/src/Rules.zifort.mk
--- open_SURFEX_V8_1_orig/src/Rules.zifort.mk	2023-03-08 16:39:38.093796000 +0100
+++ open_SURFEX_V8_1/src/Rules.zifort.mk	2023-03-08 16:40:07.892055000 +0100
@@ -89,7 +89,7 @@
 # NetCDF external
 #
 ifeq "$(VER_CDF)" "CDFEXT"
-NETCDF = $(HOME)/hm_libs/ifort/netcdf_432
+NETCDF = $(EBROOTNETCDF)
 NETCDF_INC = $(NETCDF)/include
 NETCDF_LIB = $(NETCDF)/lib
 INC_NETCDF = -I$(NETCDF_INC)
