--- a/AmberTools/src/pytraj/CMakeLists.txt
+++ b/AmberTools/src/pytraj/CMakeLists.txt
@@ -66,7 +66,6 @@
     \"CXX=${CMAKE_CXX_COMPILER}\"
     \"CC=${CMAKE_C_COMPILER}\"
     ${RPATH_ARG}
-    ${PYTHONPATH_SET_CMD}
 	${PYTHON_EXECUTABLE} ./setup.py build -b ${BUILD_DIR} ${OPENMP_ARG} ${PYTHON_COMPILER_ARG} ${WIN64_DEFINE_ARG} install -f ${PYTHON_PREFIX_ARG} --single-version-externally-managed --root /
     WORKING_DIRECTORY \"${CMAKE_CURRENT_SOURCE_DIR}\")")
 
