--- a/AmberTools/src/cpptraj/test/Test_General/RunTest.sh
+++ b/AmberTools/src/cpptraj/test/Test_General/RunTest.sh
@@ -5,7 +5,7 @@
 # Clean
 CleanFiles general.in distance.dat rmsd.dat rmsda.dat phi2.dat PhiPsi.dat \
            test.crd a1.dat Restart/* Restart test.nc r4.dat a2.dat.gz \
-           a3.dat.bz2 r2.dat r3-nofit.dat r5.dat
+           a2.dat a2.dat.save a3.dat.bz2 r2.dat r3-nofit.dat r5.dat
 
 TESTNAME='General tests'
 # Required environment 
@@ -76,7 +76,9 @@
 UNITNAME='Gzipped output data file comparison'
 CheckFor testos Linux
 if [ $? -eq 0 ] ; then
-  DoTest a2.dat.gz.save a2.dat.gz
+  gzip -cd a2.dat.gz.save > a2.dat.save
+  gzip -cd a2.dat.gz > a2.dat
+  DoTest a2.dat.save a2.dat
 fi
 DoTest a3.dat.bz2.save a3.dat.bz2
 DoTest r2.dat.save r2.dat