diff -ruN louvain-0.8.0_orig/setup.py louvain-0.8.0/setup.py
--- louvain-0.8.0_orig/setup.py	2023-02-09 11:21:33.976337043 +0100
+++ louvain-0.8.0/setup.py	2023-02-10 13:06:11.997482500 +0100
@@ -80,6 +80,7 @@
             "/usr/lib",
             "/lib64",
             "/lib",
+            # "EBROOTGCCCORE",
         ]
     else:
         extra_libdirs = []
@@ -595,10 +596,10 @@
         build_folder = vendor_folder / "build" / "igraph"
         install_folder = vendor_folder / "install" / "igraph"
 
-        if install_folder.exists():
-            # Vendored igraph already compiled and installed, just use it
-            self.use_vendored_igraph()
-            return True
+        # if install_folder.exists():
+        #     # Vendored igraph already compiled and installed, just use it
+        self.use_vendored_igraph()
+        return True
 
         if (source_folder / "CMakeLists.txt").exists():
             igraph_builder = IgraphCCoreCMakeBuilder()
@@ -648,6 +649,7 @@
         ext.runtime_library_dirs = self.runtime_library_dirs
         ext.libraries = self.libraries
         ext.extra_compile_args = self.extra_compile_args
+        print("XXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXX")
         ext.extra_link_args = self.extra_link_args
         ext.extra_objects = self.extra_objects
         ext.define_macros = self.define_macros
@@ -745,7 +747,7 @@
         the include and library paths and the library names accordingly."""
         building_on_windows = building_on_windows_msvc()
 
-        vendor_dir = Path("vendor") / "install" / "igraph"
+        vendor_dir = Path("EBROOTIGRAPH")
 
         buildcfg.include_dirs = [str(vendor_dir / "include" / "igraph")]
         buildcfg.library_dirs = []
