--- tools/rscripts/demulti.R.orig	2025-05-08 14:20:44.568946000 +0200
+++ tools/rscripts/demulti.R	2025-05-08 14:21:24.980700000 +0200
@@ -267,7 +267,7 @@
   # Sequence variability accounted for and ambiguous, degenerate, and unassigned
   # sequences identified
   degenerate_idxs <- which(
-    stringr::str_count(unique_index_seqs, "N") > max.N.count
+    stringr::str_count(as.character(unique_index_seqs), "N") > max.N.count
   )
   
   ambiguous_idxs <- as.numeric(names(table(unlist(bc_to_unique_idxs)))[
