--- ni/src/lib/nfp/eemd.h.orig	2025-12-03 11:47:58.023249000 +0100
+++ ni/src/lib/nfp/eemd.h	2025-12-03 11:48:55.346147000 +0100
@@ -16,6 +16,16 @@
  * along with libeemd.  If not, see <http://www.gnu.org/licenses/>.
  */
 
+#if !defined(__STDC_VERSION__) || __STDC_VERSION__ < 199901L
+# ifndef restrict
+#  if defined(__clang__) || defined(__GNUC__) || defined(__INTEL_COMPILER)
+#   define restrict __restrict__
+#  else
+#   define restrict
+#  endif
+# endif
+#endif
+
 #ifndef _EEMD_H_
 #define _EEMD_H_
 
